plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -826,13 +843,47 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
     @Override
     public Map<SchemaTableName, RelationType> getRelationTypes(ConnectorSession session, Optional<String> optionalSchemaName)
```

Comment:
Why is `listTables` not benefitting from similar parallelization ?
The core of both methods looks the same